spec/std/hash_spec.cr

```diff
@@ -855,6 +855,17 @@ describe "Hash" do
     h2.should be_empty
   end
 
+  it "transforms keys with values included" do
+    h1 = {1 => "a", 2 => "b", 3 => "c"}
+
+    expected_values = h1.values
+    h2 = h1.transform_keys do |x, v|
+      v.should eq(expected_values.shift)
+      x + 1
+    end
+    h2.should eq({2 => "a", 3 => "b", 4 => "c"})
```

Comment:
Food for thought: Maybe a simpler implementation could combine key and value?
```suggestion
    h1.transform_keys{ |k, v| "#{k}#{v}" }.should eq({"2a" => "a", "3b" => "b", "4c" => "c"})
```
The same can be applied to all other tests.